src/backend/server.ts

```diff
@@ -0,0 +1,89 @@
+import Fastify, { type FastifyRequest } from 'fastify';
+import { readFile } from 'node:fs/promises';
+import { config } from './lib/config.js';
+import { rootDir } from './lib/constants.js';
+import { DocumentHandler } from './lib/documentHandler.js';
+import type { FastifyRequestGeneric } from './lib/types.js';
+
+// Use dynamic imports to ensure only one of the stores gets loaded
+const preferredStore =
+	config.storage.type === 'file'
+		? new (await import('./lib/document_stores/file.js')).FileDocumentStore(config.storage)
+		: new (await import('./lib/document_stores/redis.js')).RedisDocumentStore(config.storage);
+
+// Load all static documents
+for (const [name, path] of Object.entries(config.documents)) {
+	const data = await readFile(path, 'utf-8');
+
+	if (data) {
+		await preferredStore.set(name, data, true);
+	}
+}
+
+// Setup the document handler with the preferred store
+const documentHandler = new DocumentHandler({
+	store: preferredStore,
+	keyLength: config.keyLength
+});
+
+// Initialize fastify with a restricted body size
+const fastify = Fastify({ bodyLimit: config.maxLength });
+
+// Set up the error handling for max length errors
+fastify.setErrorHandler((error, _, reply) => {
+	if (error instanceof Fastify.errorCodes.FST_ERR_CTP_BODY_TOO_LARGE) {
+		return reply.status(413).send({ message: 'Document exceeds maximum length.' });
+	}
+
+	return reply.send(error);
+});
+
+// Register the rate limit handler
+await fastify.register(import('@fastify/rate-limit'), {
+	max: 500,
+	timeWindow: '1 minute'
```

Comment:
Aren't 500 requests per minute a bit too high? Also, this could probably be customised with env variables.

Reply:
That's what it was before 
![ishare-1710452112](https://github.com/skyra-project/haste-server/assets/4019718/985fd14d-0289-40ae-b047-9ecc7dcdda5b)